README.md

```diff
@@ -1,160 +1,529 @@
-* `name` (optional): specify a name when you are using `google_auth_qr`, which will show up as the label after scanning. `[KEY NAME]` in the previous line.
+#### Getting a time-based token for the current time
 
-#### Examples
+```js
+// Generate a time-based token based on the base-32 key.
+// HOTP (counter-based tokens) can also be used if `totp` is replaced by
+// `hotp` (i.e. speakeasy.hotp()) and a `counter` is given in the options.
+var token = speakeasy.totp({
+  secret: base32secret
+});
 
-```javascript
-// generate a key
-speakeasy.generate_key({length: 20, symbols: true});
-// => { ascii: 'km^A?n&sOPJW.iCKPHKU', hex: '6b6d5e413f6e26734f504a572e69434b50484b55', base32: 'NNWV4QJ7NYTHGT2QJJLS42KDJNIEQS2V' }
+// Returns token for the secret at the current time
+// Compare this to user input
+```
 
-// generate a key and request QR code links
-speakeasy.generate_key({length: 20, qr_codes: true});
-// => { ascii: 'eV:JQ1NedJkKn&]6^i>s', ... (truncated)
-//      qr_code_ascii: 'https://www.google.com/chart?chs=166x166&chld=L|0&cht=qr&chl=eV%3AJQ1NedJkKn%26%5D6%5Ei%3Es',
-//      qr_code_hex: 'https://www.google.com/chart?chs=166x166&chld=L|0&cht=qr&chl=65563a4a51314e65644a6b4b6e265d365e693e73',
-//      qr_code_base32: 'https://www.google.com/chart?chs=166x166&chld=L|0&cht=qr&chl=MVLDUSSRGFHGKZCKNNFW4JS5GZPGSPTT' }
+#### Verifying a token
 
-// generate a key and get a QR code you can scan with the Google Authenticator app
-speakeasy.generate_key({length: 20, google_auth_qr: true});
-// => { ascii: 'V?9f6.Cq1&<H?<nxe.XJ', ... (truncated)
-//      google_auth_qr: 'https://www.google.com/chart?chs=166x166&chld=L|0&cht=qr&chl=otpauth://totp/SecretKey%3Fsecret=KY7TSZRWFZBXCMJGHRED6PDOPBSS4WCK' }
+```js
+// Verify a given token
+var tokenValidates = speakeasy.totp.verify({
+  secret: base32secret,
+  token: '123456',
+  window: 6
+});
+// Returns true if the token matches
 ```
 
-## Running the tests
-
-You'll need to have `vows` installed. If you don't have vows, `npm install -g vows`
+#### Verifying a token and calculating a delta
+
+A TOTP is incremented every `step` time-step seconds. By default, the time-step
+is 30 seconds. You may change the time-step using the `step` option, with units
+in seconds.
+
+```js
+// Verify a given token is within 3 time-steps (+/- 2 minutes) from the server
+// time-step.
+var tokenDelta = speakeasy.totp.verifyDelta({
+  secret: base32secret,
+  token: '123456',
+  window: 2,
+  step: 60
+});
+// Returns {delta: 0} where the delta is the time step difference
+// between the given token and the current time
+```
 
-To run the tests, run this in the module root directory: `vows --spec test/*`
+#### Calculating a counter-based token
 
-## Issues and patches
+```js
+var token = speakeasy.hotp({
+  secret: base32secret,
```

Comment:
Ha, that's actually exactly what I did but I removed it, but forgot to remove base32secret. In step 1, I had:

```js
current_user.two_factor_temp_secret = secret.base32
```

And step 3 had:

```js
base32secret = current_user.two_factor_temp_secret
```

But I removed all of that because I was worried it would be confusing, and replaced it with the text that it should be stored somewhere temporarily. Not sure what the right way to do it is.